src/lib/mdns/minimal/tests/TestAdvertiser.cpp

```diff
@@ -191,41 +273,199 @@ void OperationalAdverts(nlTestSuite * inSuite, void * inContext)
+    server.AddExpectedRecord(&ptrServiceSubOpenWindow);
+    NL_TEST_ASSERT(inSuite, SendQuery(kDnsSdQueryName) == CHIP_NO_ERROR);
+    NL_TEST_ASSERT(inSuite, server.GetSendCalled());
+    NL_TEST_ASSERT(inSuite, server.GetHeaderFound());
+
+    // Requests for _matter._tcp.local will respond with all records from both operational records, but no commissionable.
+    ChipLogProgress(Discovery, "Testing response to _matter._tcp.local");
+    server.Reset();
+    server.AddExpectedRecord(&ptrOperational1);
+    server.AddExpectedRecord(&srvOperational1);
+    server.AddExpectedRecord(&txtOperational1);
+    server.AddExpectedRecord(&ptrOperational2);
+    server.AddExpectedRecord(&srvOperational2);
+    server.AddExpectedRecord(&txtOperational2);
+    NL_TEST_ASSERT(inSuite, SendQuery(kMatterOperationalQueryName) == CHIP_NO_ERROR);
+    NL_TEST_ASSERT(inSuite, server.GetSendCalled());
+    NL_TEST_ASSERT(inSuite, server.GetHeaderFound());
+
+    // Responses to _matterc query should return commissionable node, but no operational.
+    ChipLogProgress(Discovery, "Testing response to _matterc._udp.local");
+    server.Reset();
+    server.AddExpectedRecord(&ptrCommissionableNode);
+    server.AddExpectedRecord(&srvCommissionableNode);
+    server.AddExpectedRecord(&txtCommissionableNodeParamsLarge);
+    NL_TEST_ASSERT(inSuite, SendQuery(kMatterCommissionableNodeQueryName) == CHIP_NO_ERROR);
+    NL_TEST_ASSERT(inSuite, server.GetSendCalled());
+    NL_TEST_ASSERT(inSuite, server.GetHeaderFound());
+
+    // Requests for each SRV record should give only records specific to that fabric.
+    ChipLogProgress(Discovery, "Testing response to operational instance name for fabric 1");
+    server.Reset();
+    // Just the SRV and TXT should return
+    server.AddExpectedRecord(&srvOperational1);
+    server.AddExpectedRecord(&txtOperational1);
+    NL_TEST_ASSERT(inSuite, SendQuery(kInstanceName1) == CHIP_NO_ERROR);
+    NL_TEST_ASSERT(inSuite, server.GetSendCalled());
+    NL_TEST_ASSERT(inSuite, server.GetHeaderFound());
+
+    ChipLogProgress(Discovery, "Testing response to operational instance name for fabric 2");
+    server.Reset();
+    // Just the SRV and TXT should return
+    server.AddExpectedRecord(&srvOperational2);
+    server.AddExpectedRecord(&txtOperational2);
+    NL_TEST_ASSERT(inSuite, SendQuery(kInstanceName2) == CHIP_NO_ERROR);
+    NL_TEST_ASSERT(inSuite, server.GetSendCalled());
+    NL_TEST_ASSERT(inSuite, server.GetHeaderFound());
+
+    ChipLogProgress(Discovery, "Testing response to commissionable instance name");
+    server.Reset();
+    // Just the SRV and TXT should return
+    server.AddExpectedRecord(&srvCommissionableNode);
+    server.AddExpectedRecord(&txtCommissionableNodeParamsLarge);
+    NL_TEST_ASSERT(inSuite, SendQuery(instanceName) == CHIP_NO_ERROR);
+    NL_TEST_ASSERT(inSuite, server.GetSendCalled());
+    NL_TEST_ASSERT(inSuite, server.GetHeaderFound());
+
+    // NL_TEST_ASSERT(inSuite, true == false);
```

Comment:
delete the debug code

Reply:
whoops, ha - now we all know the weird way I check to make sure the tests are running. Deleted in next commit.